src/SupabaseClient.ts

```diff
@@ -95,11 +96,24 @@ export default class SupabaseClient<
     this.storageKey = settings.auth.storageKey ?? ''
     this.headers = settings.global.headers ?? {}
 
-    this.auth = this._initSupabaseAuthClient(
-      settings.auth ?? {},
-      this.headers,
-      settings.global.fetch
-    )
+    if (!settings.accessToken) {
+      this.auth = this._initSupabaseAuthClient(
+        settings.auth ?? {},
+        this.headers,
+        settings.global.fetch
+      )
+    } else {
+      this.accessToken = settings.accessToken
+
+      this.auth = new Proxy<SupabaseAuthClient>({} as any, {
+        get: (prop) => {
+          throw new Error(
+            `@supabase/supabase-js: Supabase Client is configured with the accessToken option, accessing supabase.auth.${prop} is not possible`
+          )
+        },
+      })
```

Comment:
@hf why would we not want folks to access this if the access token is set?

Reply:
is it because we want to prevent supabase auth from overriding the access token set?

Reply:
No, it is because this overrides any Supabase Auth access token, so using `supabase.auth.,,,` makes no more sense, and this prevents that confusion for now.

What access token will storage receive? Not Supabase Auth's.